Native Image UnresolvedElementException sun.misc.SharedSecrets since version 3.5.2

While updating `io.projectreactor reactor-core` from version 3.5.1 to 3.5.2, the following error occurs in my [showcase-quarkus-eventsourcing](https://github.com/JohT/showcase-quarkus-eventsourcing) GraalVM [native image build (Java 17)](https://github.com/JohT/showcase-quarkus-eventsourcing/actions/runs/3899597517/jobs/6659395744):

```
Caused by: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved type during parsing: sun.misc.SharedSecrets. This error is reported at image build time because class reactor.core.publisher.Traces$SharedSecretsCallSiteSupplierFactory$TracingException is registered for linking at image build time by command line
[493](https://github.com/JohT/showcase-quarkus-eventsourcing/actions/runs/3899597517/jobs/6659395744#step:6:494)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.reportUnresolvedElement(SharedGraphBuilderPhase.java:333)
```

References: 
- [Failed Build Log](https://github.com/JohT/showcase-quarkus-eventsourcing/actions/runs/3899597517/jobs/6659395744)
- [Version Update Merge Request](https://github.com/JohT/showcase-quarkus-eventsourcing/pull/142) 
- [Related reactor-core commit](https://github.com/reactor/reactor-core/commit/9d47a04b3accfed745f8a5e7acaaf3b1b66d67e8)
- [Native image build configuration with `native-image` folder](https://www.graalvm.org/22.1/reference-manual/native-image/BuildConfiguration)

## Expected Behavior
Succeeding native image build for Java 17 that isn't influenced by features that aren't used.

## Actual Behavior
Native image build fails because of missing `sun.misc.SharedSecrets` that weren't needed prior to version 3.5.2. I suspect the newly added [reflect-config.json](https://github.com/reactor/reactor-core/blob/main/reactor-core/src/main/resources/META-INF/native-image/io.projectreactor/reactor-core/reflect-config.json), but am not sure why this is the case.

## Steps to Reproduce
See [Failed Build Log](https://github.com/JohT/showcase-quarkus-eventsourcing/actions/runs/3899597517/jobs/6659395744).
Or checkout [showcase-quarkus-eventsourcing](https://github.com/JohT/showcase-quarkus-eventsourcing) branch [renovate/patch-reactor-core.version](https://github.com/JohT/showcase-quarkus-eventsourcing/tree/renovate/patch-reactor-core.version) and follow the [instructions to build the native image](https://github.com/JohT/showcase-quarkus-eventsourcing/blob/master/showcase-quarkus-eventsourcing/README.md#-native-image).

## Possible Solution
I tried to change my [reflection-config.json](https://github.com/JohT/showcase-quarkus-eventsourcing/blob/master/showcase-quarkus-eventsourcing/reflection-config.json) and the [--exclude-config](https://github.com/graalvm/native-build-tools/issues/185). Both didn't work. I'll continue trying.

Would it be possible to find an alternative to the use of `sun.misc.SharedSecrets`? Or is there a configuration or a dependency that can be added by users of this library to get it to work with Java 17?

## Your Environment

* Reactor version(s) used:3.5.2
* Other relevant libraries versions (eg. `netty`, ...): GraalVM version 22.3.0
* JVM version (`java -version`): Java 17
* OS and version (eg `uname -a`): Ubuntu latest
* Further details see my [pipeline configuration](https://github.com/JohT/showcase-quarkus-eventsourcing/blob/renovate/patch-reactor-core.version/.github/workflows/native-image.yml)

--- a/reactor-core/src/main/resources/META-INF/native-image/io.projectreactor/reactor-core/reflect-config.json
+++ b/reactor-core/src/main/resources/META-INF/native-image/io.projectreactor/reactor-core/reflect-config.json
@@ -15,18 +15,6 @@
     "condition": {
       "typeReachable": "reactor.core.publisher.Traces"
     },
-    "name": "reactor.core.publisher.Traces$SharedSecretsCallSiteSupplierFactory",
-    "methods": [
-      {
-        "name": "<init>",
-        "parameterTypes": []
-      }
-    ]
-  },
-  {
-    "condition": {
-      "typeReachable": "reactor.core.publisher.Traces"
-    },
     "name": "reactor.core.publisher.Traces$ExceptionCallSiteSupplierFactory",
     "methods": [
       {

